backup/requirements.txt

```diff
@@ -1,2 +0,0 @@
-pyln-client>=0.12
```

Comment:
Agree... But whats the trick to tell CLN to use poetry on a plugin?

The reason I'm still using pip requirements is that when I install them `--user` (and they don't clash with deps each other) CLN is able to start it without loading a special environment.

Maybe I'm missing information, if so we also need to update `README.md` on how to run plugins correctly with poetry. Currently when I have a poetry plugin I do `poetry export --without-hashes -f requirements.txt --output requirements.txt ` and install that again via `pip install --user -r requirement.txt`.  :D

Reply:
Yeah, it's a bit tricky: if instead of a full she-bang pointing to a specific `python3` we use `/use/bin/env python3` then the plugin executable will use the `$PATH` envvar. Both `venv` and `poetry run` will prefix the `$PATH` with their `python3` location (allowing to enter and exit an env). So either `python3 -m pip install` or `python3 -m poetry install` will install things in an env. With these changes we get equivalence between the two, by disallowing poetry's virtualenv logic, and allocating a throwaway env for the test.

Long story short: use `python3 .ci/test.py [pluginname]` yo tun tests as they will be run on CI

Reply:
Thx. How does cln itself start a python plugin? Does it do a similar trick to check if it needs to load a special env or is it just running `python3 plugin.py` ?